compiler/backend/src/org/jetbrains/kotlin/codegen/FunctionCodegen.java

```diff
@@ -643,11 +643,24 @@ else if (isCompatibilityStubInDefaultImpls(functionDescriptor, context, jvmDefau
 
             if (context instanceof InlineLambdaContext) {
                 int lineNumber = -1;
-                if (strategy instanceof ClosureGenerationStrategy) {
-                    KtDeclarationWithBody declaration = ((ClosureGenerationStrategy) strategy).getDeclaration();
+                if (strategy instanceof FunctionGenerationStrategy.FunctionDefault) {
```

Comment:
Look a bug in `PsiSourceCompilerForInline` there should be ClosureGenerationStrategy if KtNamedFunction if isLambda flag passed

Reply:
Probably doesn't matter since KtNamedFunction cannot have destructuring parameters though.

Reply:
It's mostly for consistency. Because non-inlined anonymous function Generated through ClosureGenerationStrategy